src/main/java/io/cryostat/agent/remote/RecordingsContext.java

```diff
@@ -217,22 +236,81 @@ private void handleStart(HttpExchange exchange) throws IOException {
         }
     }
 
+    private void handleStopOrUpdate(HttpExchange exchange, long id) throws IOException {
+        try {
+            JFRConnection conn =
+                    jfrConnectionToolkit.connect(
+                            jfrConnectionToolkit.createServiceURL("localhost", 0));
+            IFlightRecorderService svc = conn.getService();
+            IRecordingDescriptor dsc = svc.getAvailableRecordings().stream().filter(r -> r.getId() == id).findFirst().get();
+            RecordingOptionsBuilder builder = new RecordingOptionsBuilder(conn.getService());
+            
+            InputStream body = exchange.getRequestBody();
+            JsonNode jsonMap = mapper.readTree(body);
+            Iterator<Entry<String, JsonNode>> fields = jsonMap.fields();
+
+            while (fields.hasNext()) {
+                Map.Entry<String, JsonNode> field = fields.next();
+
+                switch(field.getKey()){
+                    case "state":
+                        if ("STOPPED".equals(field.getValue().toString())) {
+                            handleStop(exchange, id);
+                            break;
+                        } else if (!"UPDATE".equals(field.getValue().toString())) {
+                            exchange.sendResponseHeaders(HttpStatus.SC_BAD_REQUEST, BODY_LENGTH_NONE);
+                        }
+                        break;
+                    case "name":
+                        builder = builder.name(field.getValue().toString());
+                        break;
+                    case "duration":
+                        builder = builder.duration(field.getValue().asLong());
+                        break;
+                    case "maxSize":
+                        builder = builder.maxSize(field.getValue().asLong());
+                        break;
+                    case "maxAge":
+                        builder = builder.maxAge(field.getValue().asLong());
+                        break;
+                    case "toDisk":
+                        builder = builder.toDisk(field.getValue().asBoolean());
+                        break;
+                    default:
+                        log.warn("Unknown recording option {}", field.getKey());
+                        exchange.sendResponseHeaders(
+                                HttpStatus.SC_METHOD_NOT_ALLOWED, BODY_LENGTH_NONE);
+                        break;
+                }
+            }
+            svc.updateRecordingOptions(dsc, builder.build());
+            exchange.sendResponseHeaders(HttpStatus.SC_CREATED, BODY_LENGTH_UNKNOWN);
```

Comment:
I would expect `CREATED` to be the successful response to a `POST`, and it would normally include a `Location` header as well.

Here I would use `200 OK` and `BODY_LENGTH_UNKNOWN` if the context is going to write back the updated recording descriptor as the response. If there is no response body then it would be `204 NO CONTENT` and `BODY_LENGTH_NONE`.